geo/geos: fix error not returning on EnsureInit if no paths are input

otan opened the pull request:
Whilst the more broad change is in another PR (#47759), we are currently
not returning an error at all if no GEOS locations are found. This is
causing issues with sqlsmith roachtests, and should make it in whilst
the discussion with the other thread is ongoing.

Refs: #47932

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47934)
<!-- Reviewable:end -->

otan committed:
geo/geos: fix error not returning on EnsureInit if no paths are input

Whilst the more broad change is in another PR (#47759), we are currently
not returning an error at all if no GEOS locations are found. This is
causing issues with sqlsmith roachtests, and should make it in whilst
the discussion with the other thread is ongoing.

Refs: #47932

Release note: None

otan commented on pkg/geo/geos/geos_unix.go:
```diff
@@ -111,7 +111,10 @@ func initGEOS(locs []string) (*C.CR_GEOS, error) {
 	}
-	return nil, errors.Wrap(err, "geos: could not find location to init GEOS")
+	if err != nil {
+		return nil, errors.Wrap(err, "geos: error during GEOS init")
```
(this if err == nil { return nil } in errors.Wrap has caught me twice now)

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47934#-:-M5YkAd6BIyfP70yJXuu:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @petermattis)



<!-- Sent from Reviewable.io -->

otan commented:
ty!

bors r=sumeerbhola
